src/AudioUnit/AudioUnit.cs

```diff
@@ -1106,13 +1106,8 @@ public AudioUnitStatus SetScheduledFiles (AudioFile[] audioFiles)
 			for (int i = 0; i < count; i++)
 				handles [i] = audioFiles [i].Handle;
 
-			GCHandle gch = GCHandle.Alloc (handles, GCHandleType.Pinned);
-			IntPtr ptr = gch.AddrOfPinnedObject ();
-
-			var status = AudioUnitSetProperty (Handle, AudioUnitPropertyIDType.ScheduledFileIDs, AudioUnitScopeType.Global, 0, ptr,  Marshal.SizeOf (ptr) * count);
-
-			gch.Free ();
-			return status;
+			fixed (IntPtr* ptr = &handles [0])
```

Comment:
I think this pattern results in an IndexOutOfRangeException is the method input is `new AudioFile[0]` as there won't be an item `0`. OTOH you can take the address of `handles` itself